Esri/arcgis-maps-sdk-swift-samples
# [Update] Improve Sample Viewer search

## Description

This PR updates the sample search bar in the Sample Viewer to include the searching of a sample's description and tags in addition to the name. Currently, the search bar performs a partial-text search on the names, a partial-text search on the descriptions, and a full-text search on the tags, in that order. The search results are divided into three sections, Name, Description, and Tags. Sections do not include samples that were found in a previous section, if any. Also, sections that have no results are hidden. If the search query is empty, a list of all the samples is displayed, sorted by name. Additionally, the query search text is bolded in the sample names and descriptions. 

## Linked Issue(s)

- `swift/issues/4230`

## How To Test

- Tap on the search bar on the main screen
- Enter a search query to see the results

## Screenshots

|Before|After|
|:-:|:-:|
|![Simulator Screen Shot - iPhone 14 Pro - 2023-07-20 at 09 39 50](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/77f99a53-8729-4c81-b807-73b843af72df)|![Simulator Screen Shot - iPhone 14 Pro - 2023-07-20 at 11 45 48](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/fb9c5d1c-d668-4530-9dc6-a98354e1baaf)|

## To Discuss

A couple things to think about:
- Is partial-text search for names, partial-text search for descirptions, and full-text search for tags okay?
- Should sections include results that were found previous section?
- Should sections with no results be hidden?
- Should bold or highlighted text be used? (See https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/228#discussion_r1283779963)

|Bold (Markdown)|Highlighted (`AttributedString`)|
|:-:|:-:|
|![Bold](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/864f1de5-acab-4607-b330-a0de827e0534)|![Highlighted](https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/34120440/faf2a670-b189-45dd-8cdc-fc735e0d3768)|
